### PR TITLE
libvirt module support

### PR DESCRIPTION
**Pull request by mbelur**

### CODE REVIEW

**Comment by moio** [comment]
@cbosdo can you have a look and review this PR please?

**Review by cbosdo — APPROVED** [review]
I am not really an expert in Virtual Host Gatherers, but from the libvirt side it seems good to me. Any reason to use a Virtual Host Gatherer given the `Virtualization` system property gives the same feature with many cool additions?

**Comment by mbelur** [comment]
> I am not really an expert in Virtual Host Gatherers, but from the libvirt side it seems good to me. Any reason to use a Virtual Host Gatherer given the `Virtualization` system property gives the same feature with many cool additions?

This work is to support using the virtual host gatherer to collect libvirt data independently of SUMA being deployed

File: virtual-host-gatherer/lib/gatherer/modules/Libvirt.py [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+        """
+
+        if not node.get('uri'):
+            raise AttributeError("Missing uri parameter in infile")
+
+        splitted_url = urllib.parse.urlsplit(node.get('uri'))
+
+        if not splitted_url.scheme:
+            raise AttributeError(
+                "Missing scheme in uri.Few examples of a valid scheme: qemu, qemu+ssh, qemu+tcp")
```
[the remark this answers is not in the document]
Comment (mbelur) [reply virtual-host-gatherer/lib/gatherer/modules/Libvirt.py]: Updated

**Review by mcalmer — CHANGES_REQUESTED** [review]
Please adapt also the spec file to create a sub package for libvirt:
https://github.com/uyuni-project/virtual-host-gatherer/blob/master/virtual-host-gatherer/virtual-host-gatherer.spec

There are a lot of examples for all the other modules.

**Comment by mbelur** [comment]
> Please adapt also the spec file to create a sub package for libvirt: https://github.com/uyuni-project/virtual-host-gatherer/blob/master/virtual-host-gatherer/virtual-host-gatherer.spec
> 
> There are a lot of examples for all the other modules.

We are planning to update the spec as part of another PR. Let me know if that's okay.

**Comment by mcalmer** [comment]
We cannot merge until this is done. We do not want to have code in the git which ends up in the source package but is not in any binary package. It is ok if you want to do it in an extra PR, but we will merge only both together.

**Comment by mbelur** [comment]
> We cannot merge until this is done. We do not want to have code in the git which ends up in the source package but is not in any binary package. It is ok if you want to do it in an extra PR, but we will merge only both together.

Updated the spec file in this PR.

**Review by mcalmer — APPROVED** [review]

**Pull request merged**